library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -51,4 +62,35 @@ object JSConverters {
       arr: scala.Array[T]): JSRichGenTraversableOnce[T] =
     new JSRichGenTraversableOnce(arr)
 
+  @inline
+  implicit def JSRichFutureThenable[A](f: Future[Thenable[A]]): JSRichFuture[A] =
+    new JSRichFuture[A](f.asInstanceOf[Future[A | Thenable[A]]])
```

Comment:
Why is a `Thenable[Thenable[A]]` unsound? The typing for `Thenable[A]` is unsound if `A <: Thenable[B]`. But `Thenable[Thenable[A]]` should be just fine if the relevant code handles the flattening correctly (and I think it should).